code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -373,4 +377,69 @@ private Map<String, String> cleanContextData(Map<String, Object> eventData) {
 		}
 		return cleanedData;
 	}
+
+	/**
+	 * Adds the following keys to the given data map:
+	 * <p>__adobe.analytics.cp</p>
+	 * <p>__adobe.analytics.contextData.a.AppId</p>
+	 * Creates required paths if they are not already present in the original data map.
+	 * This should be used only after first validating the track event has valid data from the public API.
+	 *
+	 * @param data the mutable map that will have Analytics properties added.
+	 */
+	@VisibleForTesting
+	void addAnalyticsProperties(final Map<String, Object> data) {
+		// Access to the `__adobe` map
+		Map<String, Object> adobeMap = DataReader.optTypedMap(
+			Object.class,
+			data,
+			EdgeBridgeConstants.AnalyticsKeys.ADOBE,
+			new HashMap<>()
+		);
+		if (adobeMap.isEmpty()) {
+			// If the target map is empty it could be:
+			// 1. Already existing and empty
+			// 2. Newly created as the fallback value
+			// However in both cases, it is safe to replace the existing value since there will be
+			// no data loss (this applies to all similar logic)
+			data.put(EdgeBridgeConstants.AnalyticsKeys.ADOBE, adobeMap);
+		}
```

Comment:
If there is no tracking data in the event, then we shouldn't add the AppID and cp as the only `__adobe` values. Thoughts?
cc: @emdobrin

Reply:
+1